typing/typedecl.ml

```diff
@@ -4778,6 +4792,12 @@ let report_error ppf = function
       fprintf ppf
         "@[The label %a must be mutable to be declared atomic.@]"
         Style.inline_code name
+  | Modalities_on_atomic_field name ->
+    fprintf ppf
+      "@[Non-legacy modalities are not allowed on atomic fields (here, %a).\n\
+       Use one of the modality types from the Modes module in the type of the\n\
+       field instead]"
```

Comment:
"Legacy modalities" is confusing - it could mean many things.
Specifically here, I think we will say: "Modalities of comonadic axes are not allowed on atomic fields"

Reply:
I didn't say "comonadic" since I think we're trying not to say that in documentation currently, instead I went with "future". Can you check that you're happy with the new state?